libs/MeshKernel/include/MeshKernel/BilinearInterpolationOnGriddedSamples.hpp

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright (C)  Stichting Deltares, 2011-2021.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#pragma once
+
+#include <MeshKernel/Mesh2D.hpp>
+#include <MeshKernel/MeshInterpolationInterface.hpp>
+
+namespace meshkernel
+{
+    /// @brief A class for performing bilinear interpolation on gridded samples
+    class BilinearInterpolationOnGriddedSamples : public MeshInterpolationInterface
+    {
+    public:
+        /// @brief Bilinear interpolation with constant cell size (faster because no linear search is performed for each mesh node)
+        /// @param[in] mesh The input mesh
+        /// @param[in] numColumns The number of grid columns
+        /// @param[in] numRows The number of grid rows
+        /// @param[in] xOrigin The x coordinate of the grid origin
+        /// @param[in] yOrigin The y coordinate of the grid origin
+        /// @param[in] cellSize The grid cell size
+        /// @param[in] values The values of the gridded samples
+        BilinearInterpolationOnGriddedSamples(std::shared_ptr<Mesh2D> mesh,
+                                              size_t numColumns,
+                                              size_t numRows,
+                                              double xOrigin,
+                                              double yOrigin,
+                                              double cellSize,
+                                              const std::vector<double>& values);
+
+        /// @brief Bilinear interpolation with non constant cell size (slower because linear search is performed for each mesh node)
+        /// @param[in] mesh The input mesh
+        /// @param[in] xCoordinates The x coordinates of the grid
+        /// @param[in] yCoordinates The y coordinates of the grid
+        /// @param[in] values The values of the gridded samples
+        BilinearInterpolationOnGriddedSamples(std::shared_ptr<Mesh2D> mesh,
+                                              const std::vector<double>& xCoordinates,
```

Comment:
How about std::vector<Point> const& instead. Similar to origin above. The advantage is that extension to 3D would straightforward. Otherwise, the signature of each each func or method accepting coordinates has to be changed. Do you see any advantage in this?